zrml/authorized/src/migrations.rs

```diff
@@ -0,0 +1,67 @@
+use crate::{AuthorizedOutcomeReports, Config, Outcomes, Pallet};
+use frame_support::{
+    dispatch::Weight,
+    log,
+    pallet_prelude::PhantomData,
+    traits::{Get, OnRuntimeUpgrade, StorageVersion},
+};
+
+const REQUIRED_STORAGE_VERSION: u16 = 0;
+const NEXT_STORAGE_VERSION: u16 = 1;
+
+pub struct MigrateAuthorizedStorage<T>(PhantomData<T>);
+
+// Due to problems with `storage_key_iter`, we temporarily use two maps. The old `Outcomes` will
+// no longer be used and removed at the next opportunity.
+impl<T: Config> OnRuntimeUpgrade for MigrateAuthorizedStorage<T> {
+    fn on_runtime_upgrade() -> Weight
+    where
+        T: Config,
+    {
+        if StorageVersion::get::<Pallet<T>>() != REQUIRED_STORAGE_VERSION {
```

Comment:
I think we have to add weight for one storage read in case of success.

Reply:
I think we have to add a read in any case, even if we abort, right?

Reply:
See aab45b7. One read when checking the storage version. A read and a write for each item in `Outcomes`, then a write for setting the new storage version.